subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchableHierarchies.java

```diff
@@ -121,17 +128,12 @@ private SnapshotHierarchy removeUnwatchableFileSystems(SnapshotHierarchy root, I
     }
 
     private SnapshotHierarchy invalidateUnsupportedFileSystems(SnapshotHierarchy root, Invalidator invalidator) {
-        try {
-            return watchableFileSystemDetector.detectUnsupportedFileSystems()
-                .reduce(
-                    root,
-                    (updatedRoot, fileSystem) -> invalidator.invalidate(fileSystem.getMountPoint().getAbsolutePath(), updatedRoot),
-                    nonCombining()
-                );
-        } catch (NativeException e) {
-            LOGGER.warn("Unable to list file systems to check whether they can be watched. The whole state of the virtual file system has been discarded. Reason: {}", e.getMessage());
-            return root.empty();
-        }
+        return forEachUnsupportedFileSystem(
+            root,
+            (updatedRoot, fileSystem) -> invalidator.invalidate(fileSystem.getMountPoint().getAbsolutePath(), updatedRoot),
+            "The whole state of the virtual file system has been discarded",
+            root.empty()
+        );
```

Comment:
I think this would be easier to read if the exception handling was not part of `forEachUnsupportedFileSystem()`.

Reply:
I inlined the method, since the only point of the method was to do consistent exception handling.